### ISSUE TITLE
responseType setting base on have responsed

### ISSUE THREAD

**Issue by ihoment-amu** [axios/axios#1500]
in require for download a zip file, we should responseType, but when i set a  type for responseType as arraybuffer, if this require is success , response deal with arraybuffer is right, but if this require is not success for example: it return a json object as warning message, then response deal width  arraybuffer is not true, so i think it should not set type before response . and how should i code to deal for this .

**Comment by someden** [comment]
I have the same issue. Example:
``` javascript
axios.get('/file', { responseType: 'arraybuffer' })
  .catch(function (error) {
    console.log(error.response.data); // ArrayBuffer(61797)
    // but it should be a simple object { message: '...', trace: [...], ... }
  });
```
Any ideas how to get/set proper responseType for errors?

**Comment by chinesedfan** [comment]
Closed in favor of #2434.
